source/_patterns/00-atoms/forms/text-field.mustache

```diff
@@ -9,18 +9,30 @@
   {{/label}}
   <input
       type="{{inputType}}"
-      class="text-field text-field--{{inputType}}{{#state}} text-field--{{state}}{{/state}}"
+      class="text-field text-field--{{inputType}}{{#variant}} text-field--{{variant}}{{/variant}}"
     {{#id}} id="{{id}}"{{/id}}
     {{#name}} name="{{name}}"{{/name}}
     {{#autofocus}} autofocus{{/autofocus}}
     {{#value}} value="{{value}}"{{/value}}
     {{#required}} required="required"{{/required}}
     {{#disabled}} disabled="disabled"{{/disabled}}
     {{#placeholder}} placeholder="{{placeholder}}"{{/placeholder}}
+    {{#isInvalid}} aria-invalid="true"{{/isInvalid}}
+    {{#messageGroup}} aria-describedby="{{id}}"{{/messageGroup}}
   >
 
-  {{#message}}
-    <div class="form-item__message">{{{.}}}</div>
-  {{/message}}
+  {{#messageGroup}}
```

Comment:
@thewilkybarkid swapping out `message` for `messageGroup`, separating the error message from the info message: none, either or both messages may be present. If you're happy with that approach from a data point of view I'll make the rest of the changes to support this here & in the view models.